watertap/property_models/seawater_prop_pack.py

```diff
@@ -1290,28 +1339,47 @@ def _enth_mass_phase(self):
             doc="Specific enthalpy",
         )
 
-        def rule_enth_mass_phase(b, p):  # specific enthalpy, eq. 55 and 43 in Sharqawy
+        def rule_enth_mass_phase(
+            b, p
+        ):  # specific enthalpy, eq. 25 and 26 in Nayar et al. (2016)
             t = (
                 b.temperature - 273.15 * pyunits.K
             )  # temperature in degC, but pyunits in K
             S = b.mass_frac_phase_comp[p, "TDS"]
+            s = S / 1000
+            P = b.pressure - 101325 * pyunits.Pa
+            P_MPa = pyunits.convert(P, to_units=pyunits.MPa)
+
             h_w = (
-                b.params.enth_mass_param_A1
-                + b.params.enth_mass_param_A2 * t
-                + b.params.enth_mass_param_A3 * t**2
-                + b.params.enth_mass_param_A4 * t**3
+                b.params.enth_mass_param_C1
+                + b.params.enth_mass_param_C2 * t
+                + b.params.enth_mass_param_C3 * t**2
+                + b.params.enth_mass_param_C4 * t**3
             )
-            h_sw = h_w - S * (
+            h_sw0 = h_w - s * (
                 b.params.enth_mass_param_B1
-                + b.params.enth_mass_param_B2 * S
-                + b.params.enth_mass_param_B3 * S**2
-                + b.params.enth_mass_param_B4 * S**3
+                + b.params.enth_mass_param_B2 * s
+                + b.params.enth_mass_param_B3 * s**2
+                + b.params.enth_mass_param_B4 * s**3
                 + b.params.enth_mass_param_B5 * t
                 + b.params.enth_mass_param_B6 * t**2
                 + b.params.enth_mass_param_B7 * t**3
-                + b.params.enth_mass_param_B8 * S * t
-                + b.params.enth_mass_param_B9 * S**2 * t
-                + b.params.enth_mass_param_B10 * S * t**2
+                + b.params.enth_mass_param_B8 * s * t
+                + b.params.enth_mass_param_B9 * s**2 * t
+                + b.params.enth_mass_param_B10 * s * t**2
+            )
+            h_sw = h_sw0 + P_MPa * (
+                b.params.enth_mass_param_A1
+                + b.params.enth_mass_param_A2 * t
+                + b.params.enth_mass_param_A3 * t**2
+                + b.params.enth_mass_param_A4 * t**3
+                + S
```

Comment:
I think this "s" is in `g/kg`, so should be `s` instead.

Might be better to use different names: `S_kg_kg` and `S_g_kg` instead of just "S".